numba/tests/test_parfors.py

```diff
@@ -2343,6 +2343,33 @@ def test_impl(n):
                'operators.')
         self.assertIn(msg, str(raises.exception))
 
+    @skip_parfors_unsupported
+    def test_prange_two_conditional_reductions(self):
+        # issue6414
+        def test_impl():
+            A = B = 0
+            for k in range(1):
+                if k == 2:
+                    A += 1
+                else:
+                    x = np.zeros((1, 1))
+                    if x[0, 0]:
+                        B += 1
+            return A, B
+        self.prange_tester(test_impl)
+
+    @skip_parfors_unsupported
+    def test_prange_nested_reduction1(self):
+        # issue6414
+        def test_impl():
+            A = 0
+            for k in range(1):
+                for i in range(1):
+                    if i == 0:
+                        A += 1
+            return A
+        self.prange_tester(test_impl)
```

Comment:
Does this reproduce this issue? I've tried:
```python
from numba import njit, prange
import numpy as np

@njit(parallel=True)
def test_impl():
    A = 0
    for k in prange(1):
        for i in prange(1):
            if i == 0:
                A += 1
    return A

test_impl()
```
locally on mainline and it doesn't seem to trip it?

Reply:
This was something else that I thought should be tested even if it wasn't triggered by the original error.  I should remove the issue number from that test.

Reply:
ah ok, it worked on mainline and it also worked with this patch, but reduction in nested parfors seems like a good thing to have a test for, happy to leave in with the issue number removed, thanks.